### PR TITLE
MGMT-16709: Add ignition headers to Agent CR

### PR DESCRIPTION
**Pull request by CrystalChun**
https://issues.redhat.com/browse/MGMT-16709

### CODE REVIEW

File: controllers/agentmachine_controller.go [hunk 1]
```diff
@@ -459,28 +462,32 @@ func (r *AgentMachineReconciler) processBootstrapDataSecret(ctx context.Context,
...
+			expectedPrefix := "Bearer "
+			if !strings.HasPrefix(*header.Value, expectedPrefix) {
+				log.Errorf("did not find expected prefix for bearer token in user-data secret %s", *machine.Spec.Bootstrap.DataSecretName)
+				return machineConfigPool, ignitionTokenSecretRef, ignitionEndpointHTTPHeaders, errors.New("did not find expected prefix for bearer token")
+			}
+			token = (*header.Value)[len(expectedPrefix):]
+		case "TargetConfigVersionHash":
+			ignitionEndpointHTTPHeaders["TargetConfigVersionHash"] = *header.Value
+		case "NodePool":
+			ignitionEndpointHTTPHeaders["Nodepool"] = *header.Value
```
Comment (avishayt) [inline controllers/agentmachine_controller.go:489]: Maybe we should make this code generic, to just pass all headers?
```
if header.Name == "Authorization" {
    // do stuff
} else {
    ignitionEndpointHTTPHeaders[header.Name] = *header.Value
}
```

Comment (CrystalChun) [reply controllers/agentmachine_controller.go]: Right, yes! Thank you for catching that. Modified it as suggested

**Review by avishayt — APPROVED** [review]

**Pull request merged**